### PR TITLE
osd: set default timeout 30 seconds for Mon/OSD ops

### PR DESCRIPTION
**Pull request by yuezhu**
It's found that `PGCommand` may hang when issuing PG query if the PG is in some intermediate states. To prevent it from hanging there set `rados_osd_op_timeout` and `rados_mon_op_timeout` for `go-ceph` only.

r @neurodrone

### CODE REVIEW

File: exporter.go [hunk 1]
```diff
@@ -170,6 +173,17 @@ func main() {
 			// defer Shutdown to program exit
 			defer conn.Shutdown()
 
+			// Set rados_osd_op_timeout and rados_mon_op_timeout to avoid Mon
+			// and PG command hang.
+			// See https://github.com/ceph/ceph/blob/d4872ce97a2825afcb58876559cc73aaa1862c0f/src/common/legacy_config_opts.h#L1258-L1259
+			if err := conn.SetConfigOption("rados_osd_op_timeout", strconv.FormatInt(*cephRadosOSDOpTimeout, 10)); err != nil {
```
Comment (neurodrone) [inline exporter.go]: We might need to set the config options before we issue cluster connect: https://docs.ceph.com/docs/luminous/rados/api/librados/#c.rados_conf_set

```
Pre
    rados_connect() has not been called on the cluster handle
```

File: exporter.go [hunk 2]
```diff
@@ -134,10 +135,12 @@ func (c *CephExporter) Collect(ch chan<- prometheus.Metric) {
...
 	var (
-		addr        = flag.String("telemetry.addr", ":9128", "host:port for ceph exporter")
-		metricsPath = flag.String("telemetry.path", "/metrics", "URL path for surfacing collected metrics")
-		cephConfig  = flag.String("ceph.config", "", "path to ceph config file")
-		cephUser    = flag.String("ceph.user", "admin", "Ceph user to connect to cluster.")
+		addr                  = flag.String("telemetry.addr", ":9128", "host:port for ceph exporter")
+		metricsPath           = flag.String("telemetry.path", "/metrics", "URL path for surfacing collected metrics")
+		cephConfig            = flag.String("ceph.config", "", "path to Ceph config file")
+		cephUser              = flag.String("ceph.user", "admin", "Ceph user to connect to cluster.")
+		cephRadosOSDOpTimeout = flag.Int64("ceph.rados_osd_op_timeout_sec", 30, "Ceph rados_osd_op_timeout used to contact cluster in seconds. 0 means on limit")
```
Comment (neurodrone) [inline exporter.go]: We can just expose a single timeout option for rados ops and apply it to both to keep it simple.

Comment (neurodrone) [reply exporter.go]: This field could also be `flag.Duration` instead of `flag.Int()`.

File: exporter.go (hunk 1 above)
Comment (yuezhu) [reply exporter.go]: I see

File: exporter.go (hunk 2 above)
Comment (yuezhu) [reply exporter.go]: 👍

**Pull request merged**